### PR TITLE
Handle zero correctly

### PR DESCRIPTION
**Pull request by sprunk**

### CODE REVIEW

**Comment by DeinFreund** [comment]
Why are we getting zero in the first place?

File: ZkData/Ef/Account.cs [hunk 1]
```diff
@@ -305,7 +305,7 @@ public IQueryable<Account> GetSmurfs()
         {
             var myIPs = AccountIPs.Select(x => x.IP).ToList();
             var myIDs = AccountUserIDs.Select(x => x.UserID).ToList();
-            var myInstallIDs = AccountUserIDs.Select(x => x.InstallID).ToList();
+            var myInstallIDs = AccountUserIDs.Select(x => x.InstallID).Where(x => x != "0").ToList();
             var brokenIDs = new List<Int64>() {  };
             var smurfs = new ZkDataContext().Accounts.Where(x => x.AccountID != AccountID && (x.AccountIPs.Any(y => myIPs.Contains(y.IP) && y.IP != "127.0.0.1" && y.IP != "127.0.1.1" && y.IP != "94.23.170.70" && y.IP != "78.46.100.157")
                 || x.AccountUserIDs.Where(id => !brokenIDs.Contains(id.UserID)).Any(y => myIDs.Contains(y.UserID))
```
Comment (DeinFreund) [inline ZkData/Ef/Account.cs:311]: Can you maybe also move this check from the sql query into the myIDs, myInstallIDs and my IPs? It seems to complicate the query unnecessarily.

**Review by DeinFreund — APPROVED** [review]

File: ZkData/Ef/Account.cs (hunk 1 above)
Comment (sprunk) [reply ZkData/Ef/Account.cs:311]: I was unsure how to interpret this. Did that mean
```diff
-            var myIPs = AccountIPs.Select(x => x.IP).ToList();
-            var myIDs = AccountUserIDs.Select(x => x.UserID).ToList();
+            var myIPs = AccountIPs.Select(x => x.IP).Where(x => x != "0").ToList(); // IP can't be just "0" but having the same check simplifies the query
+            var myIDs = AccountUserIDs.Select(x => x.UserID).Where(x => x != "0").ToList();
```
?

Comment (DeinFreund) [reply ZkData/Ef/Account.cs:311]: No I meant to filter my IPs by `y.IP != "127.0.0.1" && y.IP != "127.0.1.1" && y.IP != "94.23.170.70" && y.IP != "78.46.100.157"` 
 and my IDs by brokenIDs.

**Pull request merged**